### ISSUE TITLE
Docker awareness

### ISSUE THREAD

**Issue by mrubin** [johannburkard/UUID#5]
When running in docker containers, it's possible for many/all docker containers in a given system to have the same MAC address. (See https://stackoverflow.com/a/26593720/1011953) This defeats much of the entropy of this UUID v1 generation.

Is it possible to include some docker-aware options? For example, the hostname could be used instead of the MAC address. This would match to the docker container id (e.g. `8b550ab9cb7f`) when running inside the container. No two docker containers on a given host should ever have the same id, and even across hosts this id should be randomized.

**Comment by johannburkard** [comment]
Not sure about this, I think the specs were pretty clear about using the MAC address 15 years ago when I wrote UUID.

**Comment by mrubin** [comment]
Docker wasn't around 15 years ago :) The purpose of using the MAC address is that it's supposed to be unique across machines. If your applications are running in Docker containers, all their MAC addresses may be the same out of the box. This eliminates much of the entropy for the UUID generation.

I ended up modifying the code to something like:

```
try {
  File f = new File("/.dockerenv");
  if (f.exists() && !f.isDirectory()) {
    // inside a docker container, the hostname is typically its container id
    macAddress = getFirstLineOfCommand("hostname");
  } else {
    macAddress = UUIDGen.hardwareAddressLookup();
  }
} catch (Exception err) {
    // Ignored.
}
```

**Comment by johannburkard** [comment]
Can you send me the output of a few ``hostname`` invocations?

**Comment by mrubin** [comment]
Some examples:

`64e3b0fa7da8`

`be57a73c25eb`

**Comment by johannburkard** [comment]
Fixed with 97c0440bcec6092d65591090d32daa89c6ca4640.
